pointillism-port.patch

```diff
@@ -0,0 +1,123 @@
+ 
+         self.actividad = puntillism.Puntillism(self)
+ 
+-        self.build_toolbar()
+-
++        self.build_toolbar()       
++        w = PrimaryMonitor.width()
++        h = PrimaryMonitor.height() - 1 * GRID_CELL_SIZE
+         self._pygamecanvas = \
+             sugargame.canvas.PygameCanvas(
+                 self,
+                 main=self.actividad.run,
+                 modules=[pygame.display])
+-
++        self._pygamecanvas.set_size_request(w, h)
+         self.set_canvas(self._pygamecanvas)
+         self._pygamecanvas.grab_focus()
+ 
+@@ -155,19 +163,17 @@ class Activity(activity.Activity):
+     def _savebutton_cb(self, button):
+         pygame.event.post(pygame.event.Event(
+             pygame.USEREVENT, action='savebutton'))
+-
++    
++    def can_close(self):
++        self.actividad.running = False
++        self.load_image_loop = False
++        return True
++    
+     def save_image(self, image):
+-        journalobj = datastore.create()
+-        journalobj.metadata['title'] = _('Pointillism')
+-        journalobj.metadata['mime_type'] = 'image/jpeg'
+-
+-        file_path = os.path.join(
+-            os.environ['SUGAR_ACTIVITY_ROOT'], 'data', 'pointillism.jpg')
+-
+-        pygame.image.save(image, file_path)
+-        journalobj.set_file_path(file_path)
+-        datastore.write(journalobj)
+-        journalobj.destroy()
++        chooser = DesktopSaveChooser(self, filename='pointiliism.png')
++        filename = chooser.get_filename()
++        if filename:
++            pygame.image.save(image, filename)
+ 
+     def open_image(self, button):
+         pygame.event.post(pygame.event.Event(
+@@ -175,27 +181,12 @@ class Activity(activity.Activity):
+ 
+     def choose_image_from_journal_cb(self):
+         ''' Create a chooser for image objects '''
+-        self.image_id = None
+-        ##
+-        self.chooser = None
+-        try:
+-            self.chooser = ObjectChooser(
+-                parent=self, what_filter=mime.GENERIC_TYPE_IMAGE)
+-        except TypeError:
+-            self.chooser = ObjectChooser(
+-                None, self,
+-                Gtk.DialogFlags.MODAL | Gtk.DialogFlags.DESTROY_WITH_PARENT)
+-        if self.chooser is not None:
+-            try:
+-                result = self.chooser.run()
+-                if result == Gtk.ResponseType.ACCEPT:
+-                    dsobject = self.chooser.get_selected_object()
+-                    self.file_path_temp = str(dsobject.get_file_path())[:]
+-
+-            finally:
+-                self.chooser.destroy()
+-                del self.chooser
+-                return self.file_path_temp
++        self.chooser = DesktopOpenChooser(self)
++        self.chooser.add_filter('.png', 'Portable Network Graphics (.png)')
++        self.chooser.add_filter('.jpg', 'JPG Images (.jpg)')
++        self.chooser.add_filter('.jpeg', 'JPG Images (.jpeg)')
++        filepath0 = self.chooser.get_filename()
```

Comment:
If the file is not selected in the chooser, I have handled the return None errors in puntillism.py in v9.1. So that should be not an issue.
I can't continue using the camera, once you press the Open Image, as the GCI task only asked me to show a message Camera Not Found, so Walter had confirmed that I had done what the task had asked for. So..

Reply:
So, you stop all the camera pipeline before you know if you have selected an image?

Reply:
It it's a simple downstream fix, we should add it, basically not killing the video pipeline before we are sure we got the image.

Reply:
Can you asses this?